tests/test_codelist.py

```diff
@@ -90,3 +90,13 @@ def test_stray_tag_fails():
         CodeList.from_directory(
             "variable", TEST_DATA_DIR / "stray_tag" / "definitions" / "variable"
         )
+
+
+def test_end_whitespace_fails():
+    """Check that typos in a tag raises expected error"""
+
+    match = "Unexpected white space at the end of a code: scenario2"
```

Comment:
Since you are testing that the white space is correctly caught I would include it in the error message as the faulty scenario is called `'scenario2 '` and not just `'scenario2'`. If you include the suggestion from above you could test the error message like this:
```suggestion
    match = "Unexpected white space at the end of a code: 'scenario2 '"
```